src/solutions/bert_finetune_solution.py

```diff
@@ -0,0 +1,93 @@
+from pathlib import Path
+from typing import Union
+
+import pandas as pd
+import pytorch_lightning as pl
+import wandb
+from transformers import logging
+
+from src.cross_validate import CrossValidation
+from src.data_reader import load_train_test_df
+from src.model_finetuning.config import CONFIG
+from src.model_finetuning.model import BertLightningModel
+from src.model_finetuning.train import predict, train
+from src.solutions.base_solution import BaseSolution
+from src.utils import get_x_columns
+
+logging.set_verbosity_error()
+
+
+class BertFinetuningSolution(BaseSolution):
+    model = None
+
+    def __init__(self, config: dict):
+        super(BertFinetuningSolution, self).__init__()
+        self.config = config
+
+    def fit(self, X: pd.DataFrame, y: pd.DataFrame, **kwargs):
```

Comment:
You could write it as 
`def fit(self, X: pd.DataFrame, y: pd.DataFrame, val_X: pd.DataFrame, val_y: pd.DataFrame, fold: int) -> None:`

that's more readable